tests/language-feature/higher-order-functions/inference.slang

```diff
@@ -0,0 +1,21 @@
+//TEST:SIMPLE:
```

Comment:
We should avoid creating just `SIMPLE` tests. Whenever possible, prefer creating a COMPUTE test. 
With `SIMPLE` tests we at least need some file-check rules.

Reply:
done, thanks